core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/internal/ResourceRepository.java

```diff
@@ -140,10 +141,10 @@ public void run(StagingArea staging) {
 				final Set<String> resources = resourceUrisByTooling.get(toolingId).stream().map(ResourceDocument::getResourceURI).map(ResourceURI::toString).collect(Collectors.toSet());
 				
 				ResourceRequests.prepareSearchVersion()
-					.all()
+					.setLimit(10_000)
 					.filterByResources(resources)
-					.build()
-					.execute(context)
+					.stream(context)
+					.flatMap(Versions::stream)
 					.forEach(version -> {
```

Comment:
As this became a `Stream`, `forEachOrdered` should be used so we don't get caught surprised by a future version of Java that enables wider parallel processing.